tests/sentry/api/endpoints/test_organization_release_assemble.py

```diff
@@ -16,38 +16,26 @@ def setUp(self):
         self.organization = self.create_organization(owner=self.user)
         self.token = ApiToken.objects.create(user=self.user, scope_list=["project:write"])
         self.team = self.create_team(organization=self.organization)
+        self.create_team_membership(self.team, user=self.user)
```

Comment:
I'm not 100% that this is correct, since now tokens that had the ability to modify releases not part of the user's scope will lose that. But I'm pretty sure users should be using ApiKeys if they want that instead, based on how releases handles `get_projects`

Reply:
From what I see of the code currently in master, `ApiKey` requests were treated as though they had global access to the organization, and `ApiToken` requests were confined to projects their memberships or the org's projects if allow_leavejoin was enabled.

Now ApiTokens would have access to any project in the organization behaving like `ApiKey`s did. We should see what @getsentry/ecosystem thinks of this. I am a bit concerned about broadening the scope of what a user's ApiToken can be used for.

Reply:
To unstale this PR and also cause I'm taking a look a month later, I wasn't sure why I was treating setting `has_valid_api_key` for `ApiToken`s so I've removed that.

But ApiToken's will still lose the defaulted `include_all_accessible` which defaulted them access to all_projects for releases that they had before.

Reply:
For context since this conversation happened in slack, user keys will be changed where they'll only have access to releases belonging projects that belong to their team. But this will match other endpoints where they're scoped to their teams already.
And for integration tokens (tokens where `is_sentry_app=True`) they'll have access to all projects for that org. Tested [here](https://github.com/getsentry/sentry/pull/22703/files#diff-da1f5429ad761ce86047c6edd528cf45d4014a56f2c64997be22f558d6ce8234R237-R286)